guides/source/active_storage_overview.md

```diff
@@ -299,6 +299,16 @@ Call `images.attached?` to determine whether a particular message has any images
 @message.images.attached?
 ```
 
+### Attaching File/IO Objects
+
+If you want to programmatically add files from other sources, like
+user-submitted URLs, files from disk or generated PDF-documents as well as in
+unit-tests, you have to pass the file as keyword arguments:
+
+```ruby
+@message.image.attach(io: File.open('/path/to/file'), filename: 'file.pdf')
```

Comment:
To be clear, I think you should append that text to the section, like this:

> Sometimes you need to attach a file that doesn’t arrive via an HTTP request. For example, you may want to attach a file you generated on disk or downloaded from a user-submitted URL. You may also want to attach a fixture file in a model test. To do that, provide a Hash containing at least an open IO object and a filename:
>
> ```ruby
> @message.image.attach(io: File.open('/path/to/file'), filename: 'file.pdf')
> ```
>
> When possible, provide a content type as well. Active Storage attempts to determine a file’s content type from its data. It falls back to the content type you provide if it can’t do that.
>
> ```ruby
> @message.image.attach(io: File.open('/path/to/file'), filename: 'file.pdf', content_type: 'application/pdf')
> ```
>
> If you don’t provide a content type and Active Storage can’t determine the file’s content type automatically, it defaults to `application/octet-stream`.